enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+3. Provide a no hassle workflow for operators to consume the trust bundle on-par with [proxy][egress-proxy-proposal]
+
+### Non-Goals
+
+None at this time.
+
+## Proposal
+
+1. openshift-installer allows the users to provide custom trust bundle for infrastructure APIs using the `install-config.yaml`.
+2. [Infrastructure][openshift-api-infrastructure] object allows user to provide custom trust in the form of a ConfigMap in `openshift-config` namespace.
+3. A controller validates the user-provided ConfigMap and provides it as an API for consumption in the form of a ConfigMap in `openshift-config-managed` namespace.
+4. The [proxy controller][cno-proxy-controller] uses the above ConfigMap another source for generating the `openshift-config-managed/trusted-ca-bundle` so that operators get the infrastructure APIs trust when using the [`config.openshift.io/inject-trusted-cabundle`][egress-proxy-proposal] mechanism.
+
+### User Stories
+
+#### AWS GovCloud testing using emulation
+
+#### AWS C2S/SC2S
+
+#### Azure Stack Hub
+
+### Risks and Mitigations
+
+None
+
+## Design Details
+
+### Can we configure the custom trust for SDKs easily?
+
+There are 2 types of ways we would need to configure custom trust for infrastructure APIs,
+
+- The operator itself is talking to the APIs so it needs to configure the SDK.
+- The operator is managing a component that communicates with infrastructure APIs so it needs to configure that component to use the custom trust.
+
+#### SDKs directly
+
+This is not trivial, as some Go SDKs require the user to configure the http.Client with appropiate transport like for example AWS and GCP, [for example](https://github.com/aws/aws-sdk-go/issues/2404#issuecomment-454196045). While there are Go SDKs like for Azure that do not provide any great way, [for example](https://github.com/Azure/azure-sdk-for-go/issues/5974).
+
+Also since configuring the trust at best requires creating custom http transport, this is difficult and error prone to the same extent that made us move to using PROXY environment variable instead of managing transports for http clients with these specific values. Using universal system trust mechanism is more safer for the same reasons.
+
+#### Components that communicate with APIs
+
+When operators manage external components the operators are bound to API provided by the component to configure the trust and more often these components choose system trust's known locations as the API, a big example is the kube-controller-manager which does not provide any API to configure the trust for cloud APIs. In this scenario the operators would have to provide management of system trust in component's container at runtime so that cloud APIs are trusted and also other endpoints it might communicate with. This is also tedious and error prone and can sometimes be very difficult if the container images have in compatible permissions to execute certain commands like `update-system-trust`.
```

Comment:
If we need cloud provider features for configuring trust and/or mechanisms on individual binaries to configure trust, we should pursue that instead of degrading the trust and security stance of our clusters.  Especially in cases where the customer made these choices specifically for security related reasons.